src/main/java/net/sf/mzmine/parameters/parametertypes/WindowSettingsParameter.java

```diff
@@ -162,6 +161,32 @@ public void applySettingsToWindow(JFrame frame) {
     if (isMaximized) {
       frame.setExtendedState(Frame.MAXIMIZED_HORIZ | Frame.MAXIMIZED_VERT);
     }
+
+    // when still outside of screen
+    // e.g. changing from 2 screens to one
+    if (!isOnScreen(frame)) {
+      // Maximise on screen 1
+      frame.setLocation(0, 0);
+      frame.setSize(1024, 800);
+      frame.setExtendedState(Frame.MAXIMIZED_HORIZ | Frame.MAXIMIZED_VERT);
+    }
+  }
+
+
+
+  public boolean isOnScreen(JFrame frame) {

```

Comment:
Make this private?

Reply:
Could also become a util method and be moved to a WindowUtil class? Or public static

Reply:
Could be, but since there is no other use for it now, I think you can keep it here as a private method